src/bindings/python/src/pyopenvino/graph/partial_shape.cpp

```diff
@@ -43,6 +43,26 @@ void regclass_graph_PartialShape(py::module m) {
                        :return: A PartialShape with the given rank (or undefined rank if not provided), and all dimensions are dynamic.
                     )");
 
+    auto check_shape_equality = [](const ov::PartialShape& self, const auto& container) {
```

Comment:
OpenVINO uses C++11, doesn't `auto` as an argument type come with C++14 onwards?

If it does, this needs to be a template util function. Or, since you define this lambda twice in `partial_shape.cpp` and `shape.cpp`, just use explicit type.

Reply:
Please,  use template function instead.